Fix running profiler in CI on the master branch

jmythms opened the pull request:
Fix #453 

Profiler was not running on CI on the master branch. Fixed comparison string value in the bad line. Also, add printing out the current branch name to console for logs. 

[This](https://github.com/trynthink/scout/actions/runs/12890537856/job/35940718317) is the current latest CI run on master showing the branch name:
![image](https://github.com/user-attachments/assets/a537c7ed-e02d-45b6-a23d-6cc74b82fca4)

aspeake commented:
Are you able to confirm that this would actually get the right branch name? I wonder because on this branch, the branch name evaluates to `branch_name="454/merge"`, where I would have expected it to `ci_profiler_fix`.

jmythms commented:
> Are you able to confirm that this would actually get the right branch name? I wonder because on this branch, the branch name evaluates to `branch_name="454/merge"`, where I would have expected it to `ci_profiler_fix`.

I can confirm it, reasons being:

1. On the GitHub documentation [here](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs). In the latest CI run for this PR, `branch_name="${{ github.ref_name }}"` evaluated to `"454/merge"` on this branch because it is a pull request:
![image](https://github.com/user-attachments/assets/1aa37d31-07fd-44fb-bfd7-721f6db8f199)

2. The [current latest CI run](https://github.com/trynthink/scout/actions/runs/12890537856/job/35940718317) on master branch evaluated `branch_name="master"`:
![image](https://github.com/user-attachments/assets/e1747f01-0277-4c42-9f26-772585fc7cc4)

aspeake commented:
> > Are you able to confirm that this would actually get the right branch name? I wonder because on this branch, the branch name evaluates to `branch_name="454/merge"`, where I would have expected it to `ci_profiler_fix`.
> 
> I can confirm it, reasons being:
> 
>     1. On the GitHub documentation [here](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs). In the latest CI run for this PR, `branch_name="${{ github.ref_name }}"` evaluated to `"454/merge"` on this branch because it is a pull request:
>        ![image](https://private-user-images.githubusercontent.com/45446967/406491093-1aa37d31-07fd-44fb-bfd7-721f6db8f199.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc3NDIwNTEsIm5iZiI6MTczNzc0MTc1MSwicGF0aCI6Ii80NTQ0Njk2Ny80MDY0OTEwOTMtMWFhMzdkMzEtMDdmZC00NGZiLWJmZDctNzIxZjZkYjhmMTk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI0VDE4MDIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4NjIxMjUyOTY0MmFkNWMyM2MwZTY5YzlmZDRhZWEyY2U1OGYyMDVjNjY4ZWRlZmE5MmJhMzJmM2JkZGVjNTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5K2-gwC-z_Z68-HaxGRoJ1XP_Alf9_Yep4hHmI7IY3Y)
> 
>     2. The [current latest CI run](https://github.com/trynthink/scout/actions/runs/12890537856/job/35940718317) on master branch evaluated `branch_name="master"`:
>        ![image](https://private-user-images.githubusercontent.com/45446967/406492825-e1747f01-0277-4c42-9f26-772585fc7cc4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc3NDIwNTEsIm5iZiI6MTczNzc0MTc1MSwicGF0aCI6Ii80NTQ0Njk2Ny80MDY0OTI4MjUtZTE3NDdmMDEtMDI3Ny00YzQyLTlmMjYtNzcyNTg1ZmM3Y2M0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI0VDE4MDIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1YWM1NGU2OGQwZDhjZGUzMDg3MjY4YjgyYTQ1NWQ2ZDA5NmVjZGQ5Y2ZkNWZkNTAyMDYwMzY0YzRmYjYwMDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t3kRynwmu07qcuXtkV2Y99SpB5CVeceY6bftOKYecIE)

So there are a couple of routes to running integration tests that might be affecting this value. It only runs when the branch is a PR to master, but can be triggered on a PR, or it can be triggered if something is _committed_ to an active PR, I wonder if the branch name ends up being different for those. Seems like it could be the reason for the last failure.

jmythms commented:
> So there are a couple of routes to running integration tests that might be affecting this value. It only runs when the branch is a PR to master, but can be triggered on a PR, or it can be triggered if something is _committed_ to an active PR, I wonder if the branch name ends up being different for those. Seems like it could be the reason for the last failure.

@aspeake is right that the previous failure was caused because a pull request triggered the CI run. The branch name was `branch_name="454/merge"` because, under the hood, GitHub creates an "ephemeral" branch named `<PR number>/merge` for this. We need the full ref to pull this branch, not the shortened branch name (see image below). Which is why the pull, and last test failed. I have changed it so that the full branch_ref is used for pulls/checks now. 

![image](https://github.com/user-attachments/assets/3eb18645-874f-4fea-8628-536309c8be70)

actions/upload-artifact v3 was deprecated and throws an error after Jan 30 2025, so updated that to v4. I also found that the tests don't fail if the branch_name was empty, for some reason. I added a check for that as well.

jmythms commented on .github/workflows/integration_tests.yml:
```diff
@@ -50,8 +50,12 @@ jobs:
-          branch_name="${{ github.ref_name }}"
-          if [[ $branch_name == 'refs/heads/master' ]]; then
+          branch_ref="${{ github.ref }}"
+              if [ -z "$branch_ref" ]; then
+                echo "Error: $branch_ref is empty. This workflow requires a valid branch reference."
```
Find out why the empty branch_ref happened? @jmythms
